src/main/java/org/whispersystems/textsecuregcm/auth/AuthorizationTokenGenerator.java

```diff
@@ -17,23 +18,26 @@
 
   private final Logger logger = LoggerFactory.getLogger(AuthorizationTokenGenerator.class);
 
-  private final byte[] key;
+  private final byte[]           key;
+  private final Optional<byte[]> userIdKey;
 
-  public AuthorizationTokenGenerator(byte[] key) {
-    this.key = key;
+  public AuthorizationTokenGenerator(byte[] key, Optional<byte[]> userIdKey) {
```

Comment:
DirectoryCredentialsGenerator?

Reply:
done in 0973896e934a0a5e4355b99e041362f437ecb162